res/controllers/Pioneer-DDJ-400-script.js

```diff
@@ -0,0 +1,741 @@
+        },
+    },
+    deck2: {
+        vuMeter: {
+            status: 0xB0,
+            data1: 0x02,
+        },
+        playPause: {
+            status: 0x91,
+            data1: 0x0B,
+        },
+        shiftPlayPause: {
+            status: 0x91,
+            data1: 0x47,
+        },
+        cue: {
+            status: 0x91,
+            data1: 0x0C,
+        },
+        shiftCue: {
+            status: 0x91,
+            data1: 0x48,
+        },
+    },
+};
+
+// Store timer IDs
+PioneerDDJ400.timers = {};
+
+// Stores padmode state and encapsulates LED logic
+PioneerDDJ400.performancePads = {
+    modes: {
+        HOTCUE: 0x1B,
+        BEATLOOP: 0x6D,
+        BEATJUMP: 0x20,
+        SAMPLER: 0x22,
+        KEYBOARD: 0x69,
+        PADFX1: 0x1E,
+        PADFX2: 0x6B,
+        KEYSHIFT: 0x6F
+    },
+
+    // Start in hotcue mode
+    state: [
+        0x1B, // Channel 1
+        0x1B  // Channel 2
+    ],
+
+    // Toggles the shift specific lights for the current mode
+    toggleShiftLights: function(channel, value) {
+        if (this.state[channel] === this.modes.BEATJUMP) {
+            midi.sendShortMsg(channel === 0 ? 0x98 : 0x9A, 0x26, value);
+            midi.sendShortMsg(channel === 0 ? 0x98 : 0x9A, 0x27, value);
+        }
+    },
+
+    // Clear all pad mode lights and reset back to hotcue state
+    reset: function() {
+        for (var m in this.modes) {
+            if (this.modes[m] !== this.modes.HOTCUE) {
+                midi.sendShortMsg(0x90, this.modes[m], 0x00);
+                midi.sendShortMsg(0x91, this.modes[m], 0x00);
+            }
+        }
+        this.state[0] = this.state[1] = this.modes.HOTCUE;
+
+        midi.sendShortMsg(0x90, this.modes.HOTCUE, 0x7f);
+        midi.sendShortMsg(0x91, this.modes.HOTCUE, 0x7f);
+    }
+};
+
+// Save the Shift State
+PioneerDDJ400.shiftState = [0, 0];
+
+// JogWheel
+PioneerDDJ400.vinylMode = true;
+PioneerDDJ400.alpha = 1.0/8;
+PioneerDDJ400.beta = PioneerDDJ400.alpha/32;
+PioneerDDJ400.highspeedScale = 150; // multiplier for fast seek through track using SHIFT+JOGWHEEL
+PioneerDDJ400.bendScale = 0.9;
+
+PioneerDDJ400.pointJumpSpace = 0.005; // amount in percent of the Song we can jump back to previous Cue or loop point
+
+PioneerDDJ400.tempoRanges = [0.06, 0.10, 0.16, 0.25]; // WIDE = 25%?
+
+// Jog wheel loop adjust
+PioneerDDJ400.loopAdjustIn = [false, false];
+PioneerDDJ400.loopAdjustOut = [false, false];
+PioneerDDJ400.loopAdjustMultiply = 50;
+
+// Beatjump pad (beatjump_size values)
+PioneerDDJ400.beatjumpPad = {
+    0x20: -1, // PAD 1
+    0x21: 1,  // PAD 2
+    0x22: -2, // PAD 3
+    0x23: 2,  // PAD 4
+    0x24: -4, // PAD 5
+    0x25: 4,  // PAD 6
+    0x26: -8, // PAD 7
+    0x27: 8   // PAD 8
+};
+
+// For controls that need it (e.g., tempo sliders)
+PioneerDDJ400.highResMSB = {
+    "[Channel1]": {},
+    "[Channel2]": {}
+};
+
+PioneerDDJ400.trackLoadedLED = function(value, group, _control) {
+    if (value) {
+        value = 0x7F;
```

Comment:
This reuse of the same variable for a different purpose is confusing. Define a separate local variable for the MIDI byte. I suggest calling it `midiValue`.

Reply:
don't know why this was implemented this way. simplified it considerably